### DISCUSSION TITLE
Is Atmos supported?

### DISCUSSION THREAD

**Discussion by smoothquark** [MusicPlayerDaemon/MPD#2042] [Q&A] [answered]
I have been using MPD for a while now (currently stock 0.23.12 from Debian 12 Bookworm) and been happily having dual output via HDMI to a Denon AVR and USB to my DAC. I have been playing FLAC 5.1 to the AVR via HDMI without issues. I have now been experimenting with Dolby Atmos for music via Tidal eac3 files. The files are decoded via eac3-core to 7.1, but this loses the height channels and Atmos metadata. As far as I am aware MPD pretty much decodes everything to PCM and will therefore not bitstream direct to the AVR via HDMI? I tried the fifo plugin, but I could not get it to work with MPV, and I think the fifo output is also decoded PCM? Perhaps there is a way to somehow get MPD to not play eac3 files (disable ffmpeg) but instead pass it to MPV?

**Comment by smoothquark** [comment]
I have been playing with MPD a bit more. It does not look like MPD supports Atmos via EAC3 directly. However, I have managed to encapsulate the EAC3 in a 2-channel wav and MPD now bitstreams successfully to my Denon AVR via SPDIF. Playing via USB to my DAC fails.
ffmpeg -i <file> -acodec copy -f spdif <file>.spdif
ffmpeg -f s16le -ac 2 -ar 192k -i <file>.spdif -acodec copy <file>.spdif.wav

I wonder if there is a way to specify this transcode for just the HDMI output and not the other?

**Reply by MaxKellermann** [reply, marked as answer]
You can configure per-output filters.

**Comment by smoothquark** [comment]
Ah. Great, thank you.
